xdsl/dialects/stablehlo.py

```diff
@@ -105,6 +112,93 @@ class TokenType(TypeAttribute, ParametrizedAttribute):
     name = "stablehlo.token"
 
 
+@irdl_attr_definition
+class DotAttr(ParametrizedAttribute):
+    """
+    Attribute that models the dimension information for dot.
+
+    https://github.com/openxla/stablehlo/blob/b075e948092d8a27ed0be48f4f8dbaa6df7e2e3e/stablehlo/dialect/StablehloAttrs.td#L82
+    """
+
+    name = "stablehlo.dot"
+
+    lhs_batching_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+    rhs_batching_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+    lhs_contracting_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+    rhs_contracting_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+
+    @staticmethod
+    def _print_parameter(
+        name: str, value: ArrayAttr[IntegerAttr[I64]], printer: Printer
+    ):
+        printer.print_string(f"\n{name} = [")
+        printer.print_list(
+            value.data,
+            lambda dim: printer.print_string(f"{dim.value.data}"),
```

Comment:
```suggestion
            lambda dim: printer.print(dim.value.data),
```

Reply:
Can ignore this if you want

Reply:
I actually prefer print_string to print, which is unfortunately pretty slow right now